src/lib/ui/thermostat/ThermostatCard.tsx

```diff
@@ -0,0 +1,149 @@
+import classNames from 'classnames'
+import { useState } from 'react'
+import {
+  isThermostatDevice,
+  type ThermostatDevice,
+  type ThermostatDeviceProperties,
+} from 'seamapi'
+
+import { FanIcon } from 'lib/icons/Fan.js'
+import { OffIcon } from 'lib/icons/Off.js'
+import { DeviceImage } from 'lib/ui/device/DeviceImage.js'
+import { ClimateSettingStatus } from 'lib/ui/thermostat/ClimateSettingStatus.js'
+
+interface ThermostatCardProps {
+  device: ThermostatDevice
+}
+
+export function ThermostatCard({ device }: ThermostatCardProps): JSX.Element {
+  return (
+    <div className='seam-thermostat-card'>
+      <Content device={device} />
+    </div>
+  )
+}
+
+function Content(props: { device: ThermostatDevice }): JSX.Element | null {
+  const { device } = props
+
+  const [temperatureUnit, setTemperatureUnit] = useState<
+    'fahrenheit' | 'celsius'
+  >('fahrenheit')
+
+  const toggleTemperatureScale = (): void => {
+    setTemperatureUnit(
+      temperatureUnit === 'fahrenheit' ? 'celsius' : 'fahrenheit'
+    )
+  }
+
+  if (device == null || (device != null && !isThermostatDevice(device))) {
+    return null
+  }
+
+  const {
+    temperature_fahrenheit: temperatureFahrenheit,
+    temperature_celsius: temperatureCelsius,
+    current_climate_setting: currentClimateSetting,
+    is_fan_running: isFanRunning,
+    relative_humidity: relativeHumidity,
+
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
```

Comment:
Do we have an upstream issue with the SDK types?

Reply:
Yeah, I think so. I'm not too sure what the underlying issue is. Basically, we need `ThermostatDevice` to have `properties` typed as `ThermostatDeviceProperties` (which inherits from `CommonDeviceProperties`), instead of `CommonDeviceProperties` alone. Maybe it's an issue in this repo, not sure.

Reply:
Ok. I can look into this one. Ok to have the work around in for now to unblock.